packages/data/abstract/polls_repository/lib/polls_repository.dart

```diff
@@ -20,42 +20,58 @@ abstract class PollsRepository {
 //--------------------------------------------------------------------------
   /// returns arithmetic mean of values of polls
   Map<PollModelType, double> arithmeticMean(int days) {
-    final lengthOfUserPolls = curUserPolls.length;
-
-    /// case lengthOfUserPolls ==  0
-    if (lengthOfUserPolls == 0) return null;
+    //
+    if (curUserPolls.isEmpty) return null;
 
     /// for all other cases
     /// gets a count of poll details
-    final detailsCount = curUserPolls[0].details.length;
-    int mood = 0;
+
+    /// we need to know how many other polls except mood we have
+    /// so we choose any poll and count its details
+    final anyPoll = curUserPolls.values.toList()[0];
+    final detailsCount = anyPoll.details.length;
+
+    /// then create placeholders for accamulation
```

Comment:
Accumulation of what? Could you please also specify in comment since you have already added one :)